### PR TITLE
Obj_Ice_Poly

### PR DESCRIPTION
**Pull request by petrie911**
The coolest actor? You be the judge.

### CODE REVIEW

File: src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c [hunk 1]
```diff
@@ -27,15 +31,148 @@ const ActorInit Obj_Ice_Poly_InitVars = {
...
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colliderHard.base);
+    } else {
+        Actor_Kill(&this->actor);
+    }
+    pos.x = this->actor.posRot.pos.x + this->actor.scale.x * (Math_Rand_S16Offset(15, 15) * (Math_Rand_ZeroOne() < 0.5f ? -1 : 1));
+    pos.y = this->actor.posRot.pos.y + this->actor.scale.y * Math_Rand_S16Offset(10, 90);
+    pos.z = this->actor.posRot.pos.z + this->actor.scale.z * (Math_Rand_S16Offset(15, 15) * (Math_Rand_ZeroOne() < 0.5f ? -1 : 1));
+    if ((globalCtx->gameplayFrames % 7) == 0) {
+        EffectSsKiraKira_SpawnDispersed(globalCtx, &pos, &zeroVec, &zeroVec, &sColorWhite, &sColorGray, 0x7D0, 5);
+    }
```
Comment (fig02) [inline src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c:113]: scale can be in decimal

File: src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c [hunk 2]
```diff
@@ -27,15 +31,148 @@ const ActorInit Obj_Ice_Poly_InitVars = {
...
+    accel.z = 0.0f;
+    vel.x = 0.0f;
+    vel.y = this->actor.scale.y;
+    vel.z = 0.0f;
+
+    for(i = 0; i < 2; i++) {
+        pos.x = this->actor.posRot.pos.x + this->actor.scale.x * (Math_Rand_S16Offset(20, 20) * (Math_Rand_ZeroOne() < 0.5f ? -1 : 1));
+        pos.y = this->actor.posRot.pos.y + this->actor.scale.y * Math_Rand_ZeroOne() * 50.0f;
+        pos.z = this->actor.posRot.pos.z + this->actor.scale.x * (Math_Rand_S16Offset(20, 20) * (Math_Rand_ZeroOne() < 0.5f ? -1 : 1));
+        func_8002829C(globalCtx, &pos, &vel, &accel, &sColorWhite, &sColorGray, Math_Rand_S16Offset(0x15E, 0x64) * this->actor.scale.x, this->actor.scale.x * 20.0f);
```
Comment (fig02) [inline src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c:133]: was the formatter run? this line is pretty long, Id think it would snap that down

Comment (petrie911) [reply src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c]: Apparently I forgot to. Formatted.

**Review by fig02 — APPROVED** [review]

**Review by NEstelami — APPROVED** [review]

File: src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c [hunk 3]
```diff
@@ -13,9 +13,13 @@
 void ObjIcePoly_Init(Actor* thisx, GlobalContext* globalCtx);
 void ObjIcePoly_Destroy(Actor* thisx, GlobalContext* globalCtx);
 void ObjIcePoly_Update(Actor* thisx, GlobalContext* globalCtx);
-void ObjIcePoly_Draw(Actor* thisx, GlobalContext* globalCtx);
+void ObjIcePoly_Draw(ObjIcePoly* this, GlobalContext* globalCtx);
```
Comment (Roman971) [inline src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c:16]: I suppose this isn't intentional?
```suggestion
void ObjIcePoly_Draw(Actor* thisx, GlobalContext* globalCtx);
```

File: src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c [hunk 4]
```diff
@@ -27,15 +31,156 @@ const ActorInit Obj_Ice_Poly_InitVars = {
...
+
+void ObjIcePoly_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    ObjIcePoly* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Ice_Poly/func_80B94470.s")
+void ObjIcePoly_Draw(ObjIcePoly* thisx, GlobalContext* globalCtx) {
```
Comment (petrie911) [inline src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c:168]: Also need to change this.
```suggestion
void ObjIcePoly_Draw(Actor* thisx, GlobalContext* globalCtx) {
```

**Review by Roman971 — APPROVED** [review]

**Pull request merged**